### DISCUSSION TITLE
Why does Prisma run the query 'SELECT 1' in some queries?

### DISCUSSION THREAD

**Discussion by gmencz** [prisma/prisma#5604] [Q&A] [answered]
I've noticed Prisma runs `SELECT 1` sometimes and sometimes it doesn't, it would be nice to know what "triggers" this to be ran and what it does, isn't a `SELECT 1` query by itself useless?

**Comment by ryands17** [comment, marked as answer]
Hey @gmencz 👋 

Th reason that you see `SELECT 1` in queries is because we have a 15 second time limit. so, you use a connection in your application. If the next connection is reserved in less than 15 seconds, we do not `SELECT 1` and we reset the timer for that connection, if it's more than 15 seconds the connection has been picked up, we'll `SELECT 1`.

Prisma creates a [connection pool](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-management#connection-pool) so when there's more than one connection in the pool, all of them have their own timers that are always reseted to current timestamp when taken out from the pool. So this is basically a health check of sorts.

Let me know if that makes sense :)

**Reply by eduhenke** [reply]
Is this timer configurable? I am currently facing a problem that the prisma client was sending a lot of those "SELECT 1" requests, which overload the database(64k SELECT 1 requests in 1 hour):
![image](https://user-images.githubusercontent.com/8304447/115418578-f96e7500-a1cf-11eb-8f37-70c4c51a32a5.png)

I have 2 pods each with connection limit set to 200, and this overloads our DB(2vCPU 7.5GB Cloud SQL Postgres).

I think that if I could tweak around this parameter I could reduce the number of the "SELECT 1" queries, and reduce the load on our DB.

**Reply by janpio** [reply]
These queries are only executed on already open connections @eduhenke. Making them less often will not change, that these connections are already open on your database server. Can you maybe start a new issue or discussion? Then we would investigate your problem a bit deeper. I am pretty sure we can figure out what is going on and find a better solution. Just post the link here after you created it, and I can follow up there.

**Reply by janpio** [reply]
Issue created: https://github.com/prisma/prisma/issues/6668

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
